src/app/utils/cipher-web-worker-helper.ts

```diff
@@ -0,0 +1,48 @@
+import { Observable } from 'rxjs/Observable';
+import { Subject } from 'rxjs/Subject';
+import { isString } from 'util';
+
+export enum CipherWebWorkerOperations {
+  CreateAdress = 0,
+  PrepareTransaction = 1,
+}
+
+export class CipherWebWorkerHelper {
+
+  private static readonly errPrefix = 'Error:';
+
+  private static readonly worker: Worker = new Worker('/assets/scripts/cipher-web-worker.js');
+  private static readonly activeWorks: Map<number, Subject<any>> = new Map<number, Subject<any>>();
+  private static initialized = false;
+
+  static ExcecuteWorker(operation: CipherWebWorkerOperations, data: any): Observable<any> {
+
+    if (!CipherWebWorkerHelper.initialized) {
+      CipherWebWorkerHelper.initialized = true;
+
+      CipherWebWorkerHelper.worker.addEventListener('message', (e: MessageEvent) => {
+        if (isString(e.data.result) && (e.data.result as string).startsWith(CipherWebWorkerHelper.errPrefix)) {
```

Comment:
As a proposition, we can make the method `ExcecuteWorker` less complicated to move the event listener body to the separate method. In that case it will look like:
`CipherWebWorkerHelper.worker.addEventListener('message', (e: MessageEvent) => eventListener(e.data));`